engine/flink/kafka-util/src/main/scala/pl/touk/nussknacker/engine/kafka/source/KafkaSource.scala

```diff
@@ -71,9 +80,23 @@ class KafkaSource[T](preparedTopics: List[PreparedKafkaTopic],
   //There is deserializationSchema.deserialize method which doesn't need Collector, however
   //for some reason KafkaDeserializationSchemaWrapper throws Exception when used in such way...
   //protected to make it easier for backward compatibility
-  protected def deserialize(topic: String, record: ProducerRecord[Array[Byte], Array[Byte]]): T = {
+  protected def deserializeTestData(topic: String, record: ProducerRecord[Array[Byte], Array[Byte]], offset: Int = -1): T = {
```

Comment:
I think thath we need to have some TODO here that in a future we would like to have all this things in test data. Wouldn't we?

Reply:
I left some TODOs around here. And simplified Consumer->Producer->Consumer conversions in RecordFormatter and KafkaSource.